src/java/org/apache/cassandra/tcm/PaxosBackedProcessor.java

```diff
@@ -176,7 +176,23 @@ public LogState getLocalState(Epoch start, Epoch end, boolean includeSnapshot)
     @Override
     public LogState getLogState(Epoch start, Epoch end, boolean includeSnapshot, Retry.Deadline retryPolicy)
     {
-        return DistributedMetadataLogKeyspace.getLogState(start, end, includeSnapshot);
+        while (!retryPolicy.reachedMax())
+        {
+            if (Thread.currentThread().isInterrupted())
+            {
+                Thread.currentThread().interrupt();
```

Comment:
I think I have asked about this one in the other thread, but wanted to learn why it is important to re-interrupt already interrupted thread.

Reply:
in this case I don't need to re-interrupt... I called `isInterrupted` and not `interrupted`.  One thing I find annoying with this pattern is I always need to relook at the implementation details to write proper, so some times I get lazy and just re-interrupt rather than look at the details...